### ISSUE TITLE
/api/1.x/cdns/{{ID}} gives incorrect response for non-existant IDs

### ISSUE THREAD

**Issue by ocket8888** [apache/trafficcontrol#2940]
requesting e.g. `/api/1.3/cdns/999999999999999999` will respond with `200 OK` and the response body:

```json
{
	"response": []
}
```
The proper response to a request for the details of an object that doesn't exist is `404 NOT FOUND` with an appropriate error message in the response body

**Comment by moltzaum** [comment]
Most other api endpoints are like this as well. They are like this because there isn't much of a distinction between query and path parameters. I used to think
```
curl -Lvsk -b $mc https://localhost/api/1.2/cdns/42`
{"alerts":[{"text":"not found","level":"error"}]}

curl -Lvsk -b $mc https://localhost/api/1.2/cdns?id=42
{"response":[]}
```
should have been correct, but now I think both should return a 404 on the premise that the id is unique. This would imply that other unique parameters should also return a 404.

**Comment by rawlinp** [comment]
This is possibly going to be fixed by #3098.

**Comment by mitchell852** [comment]
only applies to 1.x. closing.
